typescript/sdk/src/providers/transactions/submitter/builder/TxSubmitterBuilder.ts

```diff
@@ -0,0 +1,163 @@
+   * @param txSubmitterOrType The submitter to add to the builder
+   */
+  public add(
+    txSubmitterOrType: TxSubmitterXORType<HTX, HTR>,
+  ): TxSubmitterBuilder<HTX, HTR> {
+    switch (txSubmitterOrType.type) {
+      case TxSubmitterType.SIGNER:
+        this.currentSubmitter = new SignerTxSubmitter<HTX, HTR>(
+          this.multiProvider,
+          txSubmitterOrType.chain,
+        );
+        return this;
+      case TxSubmitterType.IMPERSONATED_ACCOUNT:
+        assert(
+          txSubmitterOrType.impersonatedAccountTxSubmitterProps,
+          'Must provide required props for impersonated account submitter.',
+        );
+        this.currentSubmitter = new ImpersonatedAccountTxSubmitter<HTX, HTR>(
+          this.multiProvider,
+          txSubmitterOrType.chain,
+          txSubmitterOrType.impersonatedAccountTxSubmitterProps,
+        );
+        return this;
+      case TxSubmitterType.GNOSIS_SAFE:
+        assert(
+          txSubmitterOrType.gnosisSafeTxSubmitterProps,
+          'Must provide required props for Gnosis safe submitter.',
+        );
+        this.currentSubmitter = new GnosisSafeTxSubmitter<HTX, HTR>(
+          this.multiProvider,
+          txSubmitterOrType.chain,
+          txSubmitterOrType.gnosisSafeTxSubmitterProps,
+        );
+        return this;
+      default:
+        this.currentSubmitter = txSubmitterOrType.submitter;
+        return this;
+    }
+  }
+
+  /**
+   * Sets the current transformer for the builder.
+   * @param txTransformerOrType The transformer to add to the builder
+   */
+  public transform(
+    txTransformerOrType: TxTransformerXORType<HTX>,
+  ): TxSubmitterBuilder<HTX, HTR> {
+    assert(
+      this.currentSubmitter,
+      'No submitter specified for which to execute the transform.',
+    );
+
+    switch (txTransformerOrType.type) {
+      case TxTransformerType.ICA:
+        assert(
+          txTransformerOrType.interchainAccountTxTransformerProps,
+          'Must provide required props for interchain account submitter.',
+        );
+        this.currentTransformer = new InterchainAccountTxTransformer<HTX>(
+          this.multiProvider,
+          this.currentSubmitter.chain,
+          txTransformerOrType.chain,
+          txTransformerOrType.interchainAccountTxTransformerProps,
+        );
+        return this;
+      default:
+        this.currentTransformer = txTransformerOrType.transformer;
+        return this;
+    }
+  }
+
+  /**
+   * Submits a set of transactions to the builder.
+   * @param hyperlaneTxs The transactions to submit
+   */
+  public async submit(hyperlaneTxs: HTX[]): Promise<HTR[]> {
+    assert(
+      this.currentSubmitter,
+      'Must specify submitter to submit transactions.',
+    );
+
+    this.logger.info(
+      `Submitting ${hyperlaneTxs.length} transactions to the ${this.currentSubmitter.txSubmitterType} submitter...`,
+    );
+
+    let transformedHyperlaneTxs = hyperlaneTxs;
+    if (this.currentTransformer) {
```

Comment:
how do we stack multiple transforms with this implementation?

Reply:
We previously could not, but this is now possible with the recent change